src/EnergyPlus/GroundHeatExchangers.cc

```diff
@@ -211,11 +469,762 @@ namespace GroundHeatExchangers {
+		int num_shortTimestepGFunctions = GFNC_shortTimestep.size();
+		for( int index_shortTS = 0; index_shortTS < num_shortTimestepGFunctions; ++index_shortTS ) {
+			GFNC_combined.push_back( GFNC_shortTimestep[index_shortTS] );
+			LNTTS_combined.push_back( LNTTS_shortTimestep[index_shortTS] );
+		}
+
+		// Add the rest of the long time-step g-functions to the combined curve
+		for ( int index_longTS = myRespFactors->GFNC.l(); index_longTS <= myRespFactors->GFNC.u(); ++index_longTS ) {
+				GFNC_combined.push_back( myRespFactors->GFNC( index_longTS ) );
+				LNTTS_combined.push_back( myRespFactors->LNTTS( index_longTS ) );
+		}
+
+		// Move combined values into right data struct
+		myRespFactors->time.deallocate();
+		myRespFactors->LNTTS.deallocate();
+		myRespFactors->GFNC.deallocate();
+
+		myRespFactors->time.dimension ( GFNC_combined.size(), 0.0 );
+		myRespFactors->LNTTS.dimension( GFNC_combined.size(), 0.0 );
+		myRespFactors->GFNC.dimension( GFNC_combined.size(), 0.0 );
+
+		for ( unsigned int index = 0; index < GFNC_combined.size(); ++index ) {
+			myRespFactors->time[index] = exp( LNTTS_combined[index] ) * t_s;
+			myRespFactors->LNTTS[index] = LNTTS_combined[index];
+			myRespFactors->GFNC[index] = GFNC_combined[index];
+		}
+
+	}
+
+	//******************************************************************************
+
+	void
+	GLHEVert::makeCache()
+	{
+		// For convenience
+		auto & d = myCacheData["Phys Data"];
+
+		d["Flow Rate"] = designFlow;
+		d["Soil k"] = soil.k;
+		d["Soil rhoCp"] = soil.rhoCp;
+		d["BH Top Depth"] = myRespFactors->props->bhTopDepth;
+		d["BH Length"] = myRespFactors->props->bhLength;
+		d["BH Diameter"] = myRespFactors->props->bhDiameter;
+		d["Grout k"] = myRespFactors->props->grout.k;
+		d["Grout rhoCp"] = myRespFactors->props->grout.rhoCp;
+		d["Pipe k"] = myRespFactors->props->pipe.k;
+		d["Pipe rhoCP"] = myRespFactors->props->pipe.rhoCp;
+		d["Pipe Diameter"] = myRespFactors->props->pipe.outDia;
+		d["Pipe Thickness"] = myRespFactors->props->pipe.thickness;
+		d["U-tube Dist"] = myRespFactors->props->bhUTubeDist;
+		d["Max Simulation Years"] = myRespFactors->maxSimYears;
+
+		int i = 0;
+		for ( auto & thisBH : myRespFactors->myBorholes ) {
+			++i;
+			auto & d_bh = d["BH Data"]["BH " + std::to_string( i )];
+			d_bh["X-Location"] = thisBH->xLoc;
+			d_bh["Y-Location"] = thisBH->yLoc;
+		}
+	}
+
+	//******************************************************************************
+
+	void
+	GLHEVert::readCache()
+	{
+		// For convenience
+		using json = nlohmann::json;
+
+		if ( !gio::file_exists( glheCacheFileName ) ) {
+			// if the file doesn't exist, there are no data to read
+			return;
+		} else {
+			// file exists -- read data and load if possible
+
+			// open file
+			std::ifstream ifs( glheCacheFileName );
+
+			// create empty json object
+			json json_in;
+
+			// read json_in data
+			try {
+				ifs >> json_in;
+				ifs.close();
+			} catch ( ... ) {
```

Comment:
Does nholmann itemize out the exception types that it throws?  It would be nice to actually list out each exception type and give a helpful message to the user for each particular error.